### ISSUE TITLE
Validation of FH Monster Stat Data (Note: monster/boss names aren't hidden)

### ISSUE THREAD

**Issue by Doctawin** [Lurkars/gloomhavensecretariat#60]
I had some free time recently, so I went through to double check that the Frosthaven monster stat data are consistent with the printed version+[Reddit Q&A thread](https://www.reddit.com/r/Gloomhaven/comments/zk712a/frosthaven_component_qa_those_weird_coin_cards/). Note that I'm vaguely familiar with basic coding in a few languages, but I'm not familiar with JSON. Therefore, I may have misinterpreted something with the JSON files. I'm also new to Github, so if this single issue report is bad form, let me know and I'll create different reports in the future. 

I would be happy to help work on the following issues, but I'll need to familiarize myself with Github and how to edit files first. Let me know if you want assistance. 

What does the binary variable "small" refer to under some monster conditions? 

List of discrepancies with the printed version+Reddit thread: 
1. Algox Icespeaker 
- [x] level 7 normal should be immune to brittle. 

2. Algox Snowspeaker 
- [x] level 0 elite need to remove push 1 from data (lines 190 - 194) 
- [x] level 7 normal Algox Snowspeaker is missing immunity to brittle

3. Algox Stormcaller
- [x] remove shield 1 from the base stats (lines 14 - 18) because level 0 has no shield 

4. Black Imp
- [x] level 2 elite need to remove range value from data (line 124)

5. City Guard
- [x] level 5 elite health should be 13 (line 178)

6. Deep Terror
- [x] level 4 normal has 7 health (line 51)

7. Flame Demon
- [x] level 2 elite has retaliate 2 (line 213)
- [x] level 3 elite has retaliate 2 (line 237)

8. Forest Imp 
- [x] level 6 elite need to remove range value from data (line 221)

9. Fracture of the Deep 
- [x] the UI for the specials in the app doesn't display correctly; the data shows separate element consumption instead what the stat card shows (i.e. "consume frost, any element, any element: summon one Lurker Clawcrusher" and "consume dark, any, any: summon one Lurker Mindsnipper) 
![Screenshot 2023-01-01 at 10-40-49 Gloomhaven Secretariat](https://user-images.githubusercontent.com/121493484/210184163-496a357c-bad5-445d-b288-6264aacc802f.png)

10. Harrower Infester
- [x] level 7 normal missing poison immunity 
- [x] level 6 elite should have 27 health (line 176)

11. Hound
- [x] level 7 elite retaliate value should be 3 (line 195)

12. Ice Wraith 
- [x] level 4 normal should have attack value 3 (line 68)
- [x] level 4 elite should have attack value 4 (line 208)
- [x] level 7 elite's retaliate should have range value of 3 (line 288)

13. Living Bones
- [x] level 7 normal should have attack value 4 (line 123)
- [x] level 5 elite should have shield value 1 (line 225)
- [x] level 6 elite should have shield value 1 (line 242)
- [x] level 7 elite should have shield value 1 (line 259)

14. Living Spirit
- [x] level 2 normal should have move value 2 (line 39)
- [x] level 2 normal should have attack value 3 (line 40)
- [x] level 2 elite should have move value 3 (line 138)

15. Lord of Chaos
- [x] should have no movement value in the base stats (line 11)
- [x] special 2 is "C/2 sparks (rounded up)" so should be a ceiling instead of floor (line 53)
- [x] special 2 typo "closes" should be "closest" (line 58)
- [x] level 5 remove movement variable (line 93)

16. Lurker Clawcrusher 
- [x] level 1 normal missing shield 1

17. Lurker Mindsnipper 
- [x] level 2 normal should have pierce value 2 (line 51) 
- [x] level 3 normal should have pierce value 3 (line 67) 
- [x] level 1 elite should have health value 7 (line 151)
- [x] level 1 elite should have move value 3 (line 152)

18. Lurker Soldier 
- [x] level 3 normal should have pierce value 1 (line 67)
- [x] level 2 elite should have health value 11 (line 192)
- [x] level 3 elite should have health value 13 (line 213)

19. Ooze 
- [x] level 4 elite should have health value 12 (line 166)
- [x] level 6 elite should have shield value 1 (line 206)

20. Orphan 
- [x] level 0 move value should be 3 (line 69)
- [x] special 1 healing value is different for each level, so each level will need to be updated with a formula; I believe the following formula accounts for all the values: ceiling(C * (1.5 + level * 0.5)) --- please double check, but if correct, this formula would just need to be added in the base stats section

21. Prince of Frost
- [x] level 0 retaliate value 1 (line 113)
- [x] level 1 retaliate value 1 (line 138)
- [x] special 1 heal value is different for each level; I believe the following formula accounts for all values: ceiling(level / 2 + 2)

22. Render 
- [x] level 1 should have move value 4 (line 126)

23. Rending Drake
- [x] level 5 elite move value 6 (line 164)

24. Savvas Icestorm 
- [x] level 3 normal attack value 2 (line 51)

25. Savvas Lavaflow 
- [x] level 1 normal health value 9 (line 19)

26. Seeker of the Abyss
- [x] special 1 typo: "target" should be "targets" 

27. Shrike Fiend 
- [x] level 5 normal move value 2 (line 76)
- [x] level 5 elite move value 3 (line 171)

28. The Collector 
- [ ] the app display of the specials is hard to parse, since scrolling is required; Is there a way to display it to get around the scrolling? 
- [x] level 6 and 7 special 2 needs to be "consume light: shield 2" 

29. Vermling Priest 
- [x] level 4 elite health value 5 (line 192)

30. Wind Demon
- [x] level 3 normal remove range line (line 54)
- [x] level 1 elite missing movement value - I think the elite movement values for levels 0 and 1 need to be officially checked, as the stat card shows movement of 4 for level 0 and 3 for level 1. Based on the Gloomhaven version of this monster, I think elite levels 0 and 1 should have move 3. 

31. Vestige of the Imprisoned God 
- [x] the retaliate values for each level should be: 1, 1, 1, 2, 2, 2, 3, 3 

I wasn't able to check the following monsters yet, but when I'm able to, I'll verify the data: 
<details>
  <summary>Spoiler names</summary>

Augmented Mindthief, Belara, Blacksmith, Brother, Chaos Lieutenant, Flaming Sword of Justice, Giant Piranha, Hateful Corpse, Hungry Maw, Lady in White, Large Vermling Scout, Lurker Conductor, Power Core, Reluctant Ghost, Scabwit, The Shimmer, Tormentor, Vanjal, Yoglang 

</details>

One missing data item that will be helpful to include is the icon that appears on some of the monster image corners :). 
Here's a list of affected monsters: 
<details>
  <summary>Spoiler for monsters with mentioned icon</summary>
 
- [x] black imp
- [x] forest imp 
- [x] hound 
- [x] lightning eel 
- [x] ooze 
- [x] piranha pig 
- [x] rending drake 
- [x] ruined machine 
- [x] snow imp 
- [x] spitting drake

  
</details>

**Comment by Lurkars** [comment]
Wow wow, thank you so much. I will try to address all of this ASAP!

**Comment by Doctawin** [comment]
Whoa, you're fast! I just got up and was seeing if I could help with updating some of the data :D 

Awesome work with the app, by the way; we really enjoy using it during our sessions. I've started going through the monster ability decks, but it'll take me a few days to finish checking those.

**Comment by Lurkars** [comment]
Would be so awesome! Most data was just typed by me looking and the leaked photos of the cards. So there are mistakes and typos for sure and it's really hard work to check everything. I really really appreciate if someone will check this. So thank you!

**Comment by Lurkars** [comment]
> One missing data item that will be helpful to include is the icon that appears on some of the monster image corners :). Here's a list of affected monsters

Regarding this: I also saw this icon, but I don't know the mechanic for it yet nor if it needs to be tracked by GHS somehow :shrug:

**Comment by Doctawin** [comment]
> > One missing data item that will be helpful to include is the icon that appears on some of the monster image corners :). Here's a list of affected monsters
> 
> Regarding this: I also saw this icon, but I don't know the mechanic for it yet nor if it needs to be tracked by GHS somehow 🤷

I haven't run into it yet, but I do know that it'll be helpful to later know which monsters have that icon :)

**Comment by Lurkars** [comment]
I now just added a "catching" flag for the monsters and the small "catching" icon is shown when hovering over name.

**Comment by Doctawin** [comment]
> I now just added a "catching" flag for the monsters and the small "catching" icon is shown when hovering over name.

Cool, that's a very clean implementation and it looks slick.

**Comment by Lurkars** [comment]
I mark this as done. The issue with the display of specials is a different issue.
